### PR TITLE
PHP Config - Suites

### PR DESCRIPTION
**Pull request by loic425**
With the current state of https://github.com/Behat/Behat/issues/1539

I think we need to create a specific branch "3.17".

Overview here https://behat--179.org.readthedocs.build/en/179/user_guide/configuration/suites.html#suite-paths

The filters implementation is still on PR review.

### CODE REVIEW

**Comment by carlos-granados** [comment]
@loic425 what is the plan, to show both yaml and php configuration? Not sure if we should just show php configuration and then mention somewhere that you can also use yaml with some examples, the same way that I am doing with attributes and annotations in this PR https://github.com/Behat/docs/pull/178

**Comment by stof** [comment]
I don't think we want to maintain separate versions of the documentation for each minor version (especially given that older minor versions don't receive updates according to our versioning policy). It is unfortunate that the documentation branch is named `3.0` and not `3.x` but renaming it would break all links (unless we [configure RTD to redirect that version](https://docs.readthedocs.io/en/stable/user-defined-redirects.html#redirecting-an-old-version-to-a-new-one))

**Comment by loic425** [comment]
@acoulton I've updated this PR with last changes.

File: user_guide/configuration/suites.rst [hunk 1]
```diff
@@ -103,22 +161,36 @@ criteria. The Gherkin parser comes bundled with a set of cool filters
...
+                ->withPaths('%paths.base%/features')
+                ->withContexts(ApiContext::class)
+        )
+    ;
+
+    return (new Config())
+        ->withProfile($profile)
+    ;
 
 .. note::
```
Comment (acoulton) [inline user_guide/configuration/suites.rst]: This note about the `@` character is no longer required

File: user_guide/configuration/suites.rst [hunk 2]
```diff
@@ -10,71 +10,129 @@ With suites you can configure Behat to test different kinds of features
...
+    use Behat\Config\Config;
+    use Behat\Config\Config\Filter\RoleFilter;
+    use Behat\Config\Profile;
+    use Behat\Config\Suite;
+
+    $profile = (new Profile('default'))
+        ->withSuite(
+            (new Suite('core_features'))
+                ->withPaths('%paths.base%/features/core')
+                ->withContexts(CoreDomainContext::class)
```
Comment (acoulton) [inline user_guide/configuration/suites.rst]: On PHP8.4, we'll be able to skip the extra `()` around all the `new XXX()` which I think helps with readability. Maybe we should use 8.4 syntax in the docs with a comment the first time it appears, like so:

```suggestion
    // Note: on PHP < 8.4 you will need to use extra `()` around the `new` objects to allow method chaining
    // e.g.  `(new Profile('default'))`, `(new Suite('core_features'))`, etc.
    $profile = new Profile('default')
        ->withSuite(
             new Suite('core_features')
                ->withPaths('%paths.base%/features/core')
                ->withContexts(CoreDomainContext::class)
```

**Review by acoulton — COMMENTED** [review]
This is looking good to me.

I agree with the others that we should only have one branch per major-version-series in the docs, and that we should just migrate the docs examples from yaml to PHP rather than showing both.

I'd suggest we add one single page for the YAML config that just shows a single example of a complete YAML file with all available options, perhaps with comments to mark which PHP config objects they equate to?

File: user_guide/configuration/suites.rst (hunk 2 above)
Comment (loic425) [reply user_guide/configuration/suites.rst]: Yes I agree.

Comment (loic425) [reply user_guide/configuration/suites.rst]: done

File: user_guide/configuration/suites.rst (hunk 1 above)
Comment (loic425) [reply user_guide/configuration/suites.rst]: removed

**Review by acoulton — APPROVED** [review]
Looks great, thanks @loic425 - I've just rebased it to pick up an unrelated linting issue that was fixed in 3.0 since you opened this.

**Review by carlos-granados — APPROVED** [review]
Thanks @loic425

**Pull request merged**